hazelcast/src/hazelcast/client/sql.cpp

```diff
@@ -828,6 +840,100 @@ sql_result::page_iterator::has_next() const
     return !*last_;
 }
 
+sql_result::page_iterator_sync::non_copyables::non_copyables(
+  page_iterator&& iter)
+  : preloaded_page_{}
+  , iter_{ std::move(iter) }
+{
+}
+
+sql_result::page_iterator_sync::page_iterator_sync(
+  page_iterator&& iter,
+  std::chrono::milliseconds timeout)
+  : block_(std::make_shared<non_copyables>(std::move(iter)))
+  , current_{ block_->iter_.next().get() }
+  , timeout_{ timeout }
+{
+    if (block_->iter_.has_next()) {
+        block_->preloaded_page_ = block_->iter_.next();
+    }
+}
+
+void
+sql_result::page_iterator_sync::set_timeout(std::chrono::milliseconds t)
+{
+    timeout_ = t;
+}
+
+std::chrono::milliseconds
+sql_result::page_iterator_sync::timeout() const
```

Comment:
I suggest this method to be private and sql_result to be a friend of this class. It seems user does not need it.

Reply:
I designed it as public. User might want to retrieve the timeout which is `set` at somewhere else. What's harm in it ?

Reply:
Ok, no harm.